src/main/java/com/gregtechceu/gtceu/integration/ae2/machine/MEStockingHatchPartMachine.java

```diff
@@ -0,0 +1,330 @@
+import com.lowdragmc.lowdraglib.syncdata.annotation.Persisted;
+import com.lowdragmc.lowdraglib.syncdata.field.ManagedFieldHolder;
+
+import net.minecraft.MethodsReturnNonnullByDefault;
+import net.minecraft.core.Direction;
+import net.minecraft.nbt.CompoundTag;
+import net.minecraft.network.chat.Component;
+import net.minecraft.server.TickTask;
+import net.minecraft.server.level.ServerLevel;
+import net.minecraft.world.InteractionHand;
+import net.minecraft.world.InteractionResult;
+import net.minecraft.world.entity.player.Player;
+import net.minecraft.world.phys.BlockHitResult;
+
+import appeng.api.config.Actionable;
+import appeng.api.networking.IGrid;
+import appeng.api.stacks.AEFluidKey;
+import appeng.api.stacks.AEKey;
+import appeng.api.stacks.GenericStack;
+import appeng.api.storage.MEStorage;
+import it.unimi.dsi.fastutil.objects.Object2LongMap;
+import lombok.Getter;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.function.Predicate;
+
+import javax.annotation.ParametersAreNonnullByDefault;
+
+@ParametersAreNonnullByDefault
+@MethodsReturnNonnullByDefault
+public class MEStockingHatchPartMachine extends MEInputHatchPartMachine implements IMEStockingPart, IAutoPullPart {
+
+    protected static final ManagedFieldHolder MANAGED_FIELD_HOLDER = new ManagedFieldHolder(
+            MEStockingHatchPartMachine.class, MEInputHatchPartMachine.MANAGED_FIELD_HOLDER);
+
+    private static final int CONFIG_SIZE = 16;
+
+    @DescSynced
+    @Persisted
+    @Getter
+    private boolean autoPull;
+
+    private Predicate<GenericStack> autoPullTest;
+
+    public MEStockingHatchPartMachine(IMachineBlockEntity holder, Object... args) {
+        super(holder, args);
+        this.autoPullTest = $ -> false;
+    }
+
+    @Override
+    protected @NotNull NotifiableFluidTank createTank(long initialCapacity, int slots, Object... args) {
+        this.aeFluidHandler = new ExportOnlyAEStockingFluidList(this, CONFIG_SIZE);
+        return this.aeFluidHandler;
+    }
+
+    @Override
+    public void autoIO() {
+        if (!this.isWorkingEnabled()) return;
+
+        if (this.updateMEStatus()) {
+            MEStorage aeNetwork = this.getMainNode().getGrid().getStorageService().getInventory();
+            for (ExportOnlyAEFluidSlot slot : aeFluidHandler.getInventory()) {
+                var config = slot.getConfig();
+                if (config == null) {
+                    slot.setStock(null);
+                } else {
+                    // Try to fill the slot
+                    var key = config.what();
+                    long result = aeNetwork.extract(key, Long.MAX_VALUE, Actionable.SIMULATE, actionSource);
+                    slot.setStock(new GenericStack(key, result));
+                    slot.onContentsChanged();
+                }
+            }
+            this.updateTankSubscription();
+        }
+
+        if (!autoPull || !(getOffsetTimer() % 100 == 0)) return;
```

Comment:
the config has a value named `updateIntervals` for how often the ME parts should update. use that instead of a hardcoded value?

Reply:
This is a separate value in MEStocking, used to represent the interval time for automatic pulling.

In NH, there is a PR that allows this value to be modified in GUI, but in CEU, this value is hard coded to 5 seconds

Reply:
I can add spacing and minimum pull settings like NH, but I don't know how to design GUI

Reply:
I see. that's fine, then.

Reply:
I am unable to perform hot swap in the GTM development environment (Unexpected: JDWP Error: 113). This has turned GUI writing into a disaster.

Reply:
yep, same, it seems that architectury really hates hotswapping, since on 1.21 it works perfectly.

Reply:
let's just leave it as is for now.

Reply:
i hate arch loom :(